certora/specs/SocialRecoveryModule.spec

```diff
@@ -158,3 +162,34 @@ rule revokeGuardianRevertIfAddressNotGuardian(env e, address otherAccount, addre
 
     assert lastReverted;
 }
+
+// This rule verifies that the guardian can always initiate recovery considering some ideal conditions.
+rule confirmRecoveryCanAlwaysBeInitiatedByGuardian(env e, address guardian, address[] newOwners, uint256 newThreshold, bool execute) {
+
+    require newThreshold > 0;
+    require newThreshold <= newOwners.length;
+
+    require e.msg.value == 0;
+    require e.msg.sender == guardian;
+    require currentContract.isGuardian(safeContract, guardian);
+    require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
+    require e.block.timestamp + currentContract.recoveryPeriod < max_uint64;
+
+    uint256 nonce = currentContract.nonce(safeContract);
+    require nonce < max_uint256;
+
+    bytes32 recoveryHash = currentContract.encodeRecoveryDataHash(safeContract, newOwners, newThreshold, nonce);
+    require currentContract.recoveryRequests[safeContract].executeAfter == 0; // This ensures that the recovery is not already initiated.
+
+    // This ensures that the required threshold is reached.
+    require currentContract.getRecoveryApprovals(safeContract, newOwners, newThreshold) == currentContract.threshold(safeContract);
+
+    currentContract.confirmRecovery@withrevert(e, safeContract, newOwners, newThreshold, execute);
+    bool isReverted = lastReverted;
+
+    assert !isReverted &&
+        currentContract.confirmedHashes[recoveryHash][e.msg.sender];
+    assert execute =>
+        to_mathint(currentContract.recoveryRequests[safeContract].executeAfter) == e.block.timestamp + currentContract.recoveryPeriod &&
+        currentContract.recoveryRequests[safeContract].newThreshold == newThreshold;
```

Comment:
Should we assert something about new owners?

Reply:
I added the length parameter check and a random index check in the owner array.